crates/data_components/src/kafka.rs

```diff
@@ -298,12 +312,17 @@ impl KafkaConsumer {
 pub struct KafkaMessage<'a, K, V> {
     consumer: &'a StreamConsumer,
     msg: BorrowedMessage<'a>,
-    key: K,
+    key: Option<K>,
```

Comment:
Key part is optional and does not always exist, for example when using simple/basic Kafka topic configuration